src/sources/statsd/mod.rs

```diff
@@ -44,9 +49,13 @@ fn statsd(addr: SocketAddr, shutdown: ShutdownSignal, out: mpsc::Sender<Event>)
 
     Box::new(
         async move {
-            let socket = UdpSocket::bind(&addr)
-                .await
-                .expect("failed to bind to udp listener socket");
+            let socket = match UdpSocket::bind(&addr).await {
+                Ok(socket) => socket,
+                Err(error) => {
+                    emit!(StatsdSocketError::bind(error));
+                    return Err(());
+                }
+            };
```

Comment:
This might be a little more succinct with `map_err` instead of `match`.

Reply:
How this can be solved with `map_err`? :thinking:

Reply:
```suggestion
            let socket = match UdpSocket::bind(&addr).map_err(|e| {
                emit!(StatsdSocketError::bind(error));
            }).await;
```

Reply:
Thanks, changed.